clang/lib/Driver/ScanAndUpdateArgs.cpp

```diff
@@ -0,0 +1,213 @@
+//===- ScanAndUpdateArgs.cpp - Util for CC1 Dependency Scanning -----------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "clang/Driver/ScanAndUpdateArgs.h"
+#include "clang/Frontend/CompilerInvocation.h"
+#include "clang/Lex/HeaderSearchOptions.h"
+#include "clang/Tooling/DependencyScanning/DependencyScanningTool.h"
```

Comment:
This won't work with shared libraries because you haven't added dependencies to `clangDriver`, which are very limited:
```
  LINK_LIBS
  clangBasic
  clangCAS
  ${system_libs}
```
This allows `clangDriver` to be run independently with very few dependencies. This is why the code used to pass a callback into the driver (before Steven cleaned it up in 6e2f765ae51a06198b03f8e48972e6d8f69e50da).

I think @benlangmuir and I previously discussed how this functionality probably belongs in `clangToolingDependencyScanning`. Then it can:
- Be implemented in clang-scan-deps for easy testing.
- Exposed in the dependency scanning APIs that Xcode uses.

WDYT?

Reply:
Makes sense! Moved `scanAndUpdateCC1InlineWithTool()` to `Tooling/DependencyScanning`.